mobile/android/android-emulator-services.ts

```diff
@@ -184,7 +184,7 @@ class AndroidEmulatorServices implements Mobile.IAndroidEmulatorServices {
 
 		// if we get here, there's at least one running emulator
 		let getNameFunction = this.$options.geny ? this.getNameFromGenymotionEmulatorId : this.getNameFromSDKEmulatorId;
-		let emulatorId = await _(runningEmulators).find(emulator => getNameFunction.apply(this, [emulator]) === image);
+		let emulatorId = await _(runningEmulators).find(async emulator => await getNameFunction.apply(this, [emulator]) === image);
```

Comment:
this code will not wait for the execution of the lambdas inside `find`
It should be something like:
```TypeScript
for (let emulator of runningEmulators) {
    const currentEmulatorName = await getNameFunction.apply(this, [emulator]);
    if (currentEmulatorName === image) {
        return emulator;
    }
}
```